File: ROID-Forum-Server/ROID-Forum-Server/Convert Server Code/GMGServer/bin/Server/Sections/Coding Section Controller/ThreadInfo.dart
library CodingThreadInfo;
import 'CommentInfo.dart';
import '../../Account Controller/AccountController.dart';

class ThreadInfo {
  int id;
  String title;
  String description;
  String owner;
  List<CommentInfo> comments;
  int commentIDs = 0;
  ThreadInfo.fromData(String _owner, int _id, String t, String d) {
    owner = _owner;
    id = _id;
    title = t;
    description = d;
    comments = new List();
  }
  ThreadInfo.fromMap(Map m) {
    owner = m["Owner"];
    id = m["ID"];
    title = m['Title'];
    description = m['Description'];
    List<Map> commentArray = m['Comments'];
    comments = new List();
    if (m['CommentIDs'] != null) {
      commentIDs = m['CommentIDs'];
    }
    //Repair comments when loading them just in case their IDs are wrong.
    for (int i = 0; i < commentArray.length; i++) {
      CommentInfo c = new CommentInfo.fromMap(commentArray[i]);
      if (getCommentForID(c.commentID) != null) {
        c.commentID = commentIDs++;
      }
      comments.add(c);
    }
  }
  Map toMap({AccountController addAvatars:null}) {
    Map m = new Map();
    m["Owner"] = owner;
    m["ID"] = id;
    m['Title'] = title;
    m['Description'] = description;
    if (addAvatars != null) {
      m['AvatarURL'] = addAvatars.getAvatarForAccount(owner);
    } else {
      m['CommentIDs'] = commentIDs;
    }
    List<Map> commentArray = new List();
    for (int i = 0; i < comments.length; i++) {
      commentArray.add(comments[i].toMap(addAvatars:addAvatars));
    }
    m['Comments'] = commentArray;
    return m;
  }
  CommentInfo getCommentForID(int commentID) {
    for (int i = 0; i < comments.length; i++) {
      if (comments[i].commentID == commentID) {
        return comments[i];
      }
    }
    return null;
  }
}

File: ROID-Forum-Server/ROID-Forum-Server/Section Controller/ThreadInfo.cs
using System;
using System.Collections.Generic;
namespace ROIDForumServer
{
    public class ThreadInfo
    {
        public int id;
        public String title;
        public String description;
        public String owner;
        public List<CommentInfo> comments;
        public int commentIDs = 0;
        public ThreadInfo(String _owner, int _id, String t, String d)
        {
            owner = _owner;
            id = _id;
            title = t;
            description = d;
            comments = new List<CommentInfo>();
        }
        public ThreadInfo(Dictionary<string, object> m)
        {
            owner = (string)m["Owner"];
            id = Convert.ToInt32(m["ID"]);
            title = (string)m["Title"];
            description = (string)m["Description"];
            List<Dictionary<string, object>> commentArray = (List<Dictionary<string, object>>)m["Comments"];
            comments = new List<CommentInfo>();
            if (m["CommentIDs"] != null)
            {
                commentIDs = Convert.ToInt32(m["CommentIDs"]);
            }
            //Repair comments when loading them just in case their IDs are wrong.
            for (int i = 0; i < commentArray.Count; i++)
            {
                CommentInfo c = new CommentInfo(commentArray[i]);
                if (getCommentForID(c.commentID) != null)
                {
                    c.commentID = commentIDs++;
                }
                comments.Add(c);
            }
        }
        public Dictionary<string, object> toMap()
        {
            Dictionary<string, object> m = new Dictionary<string, object>();
            m["Owner"] = owner;
            m["ID"] = id;
            m["Title"] = title;
            m["Description"] = description;
            m["CommentIDs"] = commentIDs;
            List<Dictionary<string, object>> commentArray = new List<Dictionary<string, object>>();
            for (int i = 0; i < comments.Count; i++)
            {
                commentArray.Add(comments[i].toMap());
            }
            m["Comments"] = commentArray;
            return m;
        }
        public CommentInfo getCommentForID(int commentID)
        {
            for (int i = 0; i < comments.Count; i++)
            {
                if (comments[i].commentID == commentID)
                {
                    return comments[i];
                }
            }
            return null;
        }
    }
}
